src/Control/Lens/Internal/Fold.hs

```diff
@@ -53,8 +53,8 @@ import qualified Data.List.NonEmpty as NonEmpty
 -- | A 'Monoid' for a 'Contravariant' 'Applicative'.
 newtype Folding f a = Folding { getFolding :: f a }
 
-instance (Contravariant f, Apply f) => Semigroup (Folding f a) where
-  Folding fr <> Folding fs = Folding (fr .> fs)
+instance (Contravariant f, Applicative f) => Semigroup (Folding f a) where
```

Comment:
Can you invert the dependency here as well? I'd rather see the `Monoid` instance occur an `Apply` constraint than have this `Semigroup` instance incur an `Applicative` constraint.

Reply:
It's used for `ifolded` default implementation (and nothing else), so this way is better.